internal/pkg/manifest/backend_app.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/template"
+)
+
+const (
+	backedAppManifestPath = "applications/backend-app/manifest.yml"
+)
+
+// BackendAppProps represents the configuration needed to create a backend application.
+type BackendAppProps struct {
+	AppProps
+	Port        uint16
+	HealthCheck *ContainerHealthCheck // Optional healthcheck configuration.
+}
+
+// BackendApp holds the configuration to create a backend application manifest.
+type BackendApp struct {
+	App              `yaml:",inline"`
+	Image            AppImageWithPort `yaml:",flow"`
+	backendAppConfig `yaml:",inline"`
+	Environments     map[string]backendAppConfig `yaml:",flow"`
+
+	parser template.Parser
+}
+
+type backendAppConfig struct {
```

Comment:
A unexposed struct with exposed elements?

Reply:
The fields have to be exposed for the `yaml` library to marshal the struct:

> Struct fields are only marshalled if they are exported (have an upper case first letter),

https://godoc.org/gopkg.in/yaml.v3